cloudinstall/async.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2015 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Async Handler
+Provides async operations for various api calls and other non-blocking
+work.
+"""
+
+import logging
+from concurrent.futures import ThreadPoolExecutor
+log = logging.getLogger("cloudinstall.async")
+
+
+class Async:
+    """ Only have 1 additional thread running at a time, makes for easier
+    debugging and predictability.
+    """
+    pool = ThreadPoolExecutor(1)
+    log.debug('pool={}'.format(pool))
```

Comment:
Bikeshed style comment: why have the separate Async class here?
You could just declare Pool as a module level variable - like this:
```python
# only have 1 addl thread at a time.... etc etc
AsyncPool = ThreadPoolExecutor(1)
log.debug("AsyncPool={}".format(AsyncPool))
```

and then in other files just do this:
```python
from cloudinstall.async import AsyncPool

AsyncPool.submit(blah)
```

Is there something I'm missing that the class adds? I feel like I'm forgetting some python module loading semantic subtlety here